docs/source/usecases/WindowsHeader.md

```diff
@@ -0,0 +1,7 @@
+# Kokkos and `Windows.h`
+
+When using Kokkos on Windows a program or library might include `windows.h`. This is problematic as this header defines two macros with the names `min` and `max` unless `NOMINMAX` is defined previously.
+The preprocessor replaces strings in the source code with the macros yielding an uninterpretable result and thus compilation fails.
+Therefore, the header `Kokkos_Core.hpp` is protected against these macros, meaning they are undefined at the beginning and redefined at the end of the header file.
+Even though definitions inside `Kokkos_Core.hpp` are protected against the macros, code outside is not.
+Thus, it is on the user to deal with the macros being defined, either by defining `NOMINMAX` before including `windows.h` (prefered) or by putting `()` around names that contain `min` or `max`.
```

Comment:
I think I was unclear. The preferred solution is compiling with `/DNOMINMAX` (MSVC) or `-DNOMINMAX` across the whole project, rather than sticking `#define NOMINMAX` in the source code.

Reply:
okay, and you would like to explicitly state that the it should be defined in the compile line? I was trying to leave this up to the user if it is done in the compile line or by includes. But I do see, that it reads as if I was advocating a `#define` in the source code which was not intended.

Reply:
I tried in [0355b4](https://github.com/kokkos/kokkos-core-wiki/pull/195/commits/0355b422be9884bb66664a174ff0f675cbb9fa92)